src/containers/SignupForm/SignupForm.js

```diff
@@ -31,36 +33,50 @@ class Signup extends Component {
       [name]: value
     })
   }
+  componentDidMount() {
+    const searchParams = new URLSearchParams(this.props.location.search)
+    this.state.code = searchParams.get('code') ? searchParams.get('code') : ''
```

Comment:
Do not mutate state directly. Use `setState()`.

Reply:
Do we need to maintain a state for `code`? Why not directly get the code inside `handleSubmit` function?